cxx/pclean/schema_helper.hh

```diff
@@ -38,3 +38,10 @@ class PCleanSchemaHelper {
   std::map<std::string, std::vector<std::string>> domains;
   std::map<std::string, std::vector<std::string>> annotated_domains;
 };
+
+// Returns original_domains, but with the elements corresponding to
+// annotated_ds elements that start with prefix moved to the front.
+std::vector<std::string> reorder_domains(
```

Comment:
Could you add a comment somewhere explaining how it works that this function, when applied to noisy relation domains, moves the domains corresponding to base relation domains not only to the front, but also in the same order as the base relation domains? I think I understand but that's an important guarantee and we should document it.

Reply:
Done.